main/src/observables/gravitational_waves.hpp

```diff
@@ -0,0 +1,195 @@
+/*
+* MIT License
+*
+* Copyright (c) 2021 CSCS, ETH Zurich
+*               2021 University of Basel
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*/
+
+/*! @file calculate gravitational waves observable
+ *
+ *  @author Lukas Schmidt
+ */
+
+#include <array>
+#include "mpi.h"
+
+#include "iobservables.hpp"
+#include "io/ifile_writer.hpp"
+
+namespace sphexa
+{
+
+//!@brief calculates the second derivative of the quadpole momentum
+template<class T, class Dataset>
+T d2QuadpoleMomentum(size_t begin, size_t end, int pos1, int pos2, Dataset& d)
+{
+    T* x        = d.x.data();
+    T* y        = d.y.data();
+    T* z        = d.z.data();
+    T* vx       = d.vx.data();
+    T* vy       = d.vy.data();
+    T* vz       = d.vz.data();
+    T* gradP_x  = d.grad_P_x.data();
+    T* gradP_y  = d.grad_P_y.data();
+    T* gradP_z  = d.grad_P_z.data();
+    T* m        = d.m.data();
+
+    T out = 0.0;
+
+    std::array<T*, 3> pos = {x, y, z};
+    std::array<T*, 3> vel = {vx, vy, vz};
+    std::array<T*, 3> gradP = {gradP_x, gradP_y, gradP_z};
+
+    if(pos1 == pos2)
+    {
+
+#pragma omp parallel for reduction(+ : out)
+        for(size_t i = begin; i < end; i++)
+        {
+            T acc1 = -gradP[pos1][i];
+            T factor1 = vx[i] * vx[i] + vy[i] * vy[i] + vz[i] * vz[i];
```

Comment:
since this is the square of the norm of the velocity, why not name it like that? e.g. `v2`?